platform/kube/inject/inject.go

```diff
@@ -200,10 +208,10 @@ func GetInitializerConfig(kube kubernetes.Interface, namespace, name string) (*C
 		c.Policy = DefaultInjectionPolicy
 	}
 	if c.Params.InitImage == "" {
-		c.Params.InitImage = InitImageName(DefaultHub, version.Info.Version)
+		c.Params.InitImage = InitImageName(DefaultHub, version.Info.Version, c.Params.DebugMode)
```

Comment:
having both InitImage and the debug mode depending on a flag which should agree when invoked is weird.
I started putting XxxImageName as method on Param (SetProxyImageName() which would use the flag but that doesn't work well with initialization of the struct and the rest of the code, an alternative would be to pass in the hub and version and flag instead of the image name but that's a bigger change)